src/promptflow/promptflow/_utils/utils.py

```diff
@@ -235,3 +236,14 @@ def resolve_dir_to_absolute(base_dir: Union[str, Path], sub_dir: Union[str, Path
         base_dir = base_dir if isinstance(base_dir, Path) else Path(base_dir)
         path = base_dir / sub_dir
     return path
+
+
+def parse_ua_to_dict(ua):
+    """Parse string user agent to dict with name as ua name and value as ua version."""
+    ua_dict = {}
+    ua_list = ua.split(" ")
+    for item in ua_list:
+        if item:
+            key, value = item.split("/")
```

Comment:
what if the item does not have /?

Reply:
Seems normalized ua will have it. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent